sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorReceiver.java

```diff
@@ -177,13 +152,18 @@ public void dispose() {
         if (!pendingUpdates.isEmpty()) {
             final List<Mono<Void>> pending = new ArrayList<>();
             final StringJoiner builder = new StringJoiner(", ");
-
             for (UpdateDispositionWorkItem workItem : pendingUpdates.values()) {
+
                 if (workItem.hasTimedout()) {
                     continue;
                 }
 
-                pending.add(workItem.getMono());
+                if (workItem.getDeliveryState() instanceof  TransactionalState) {
+                    pending.add(updateDispositionInternal(workItem.getLockToken(), Released.getInstance()));
+                    //pending.add(workItem.getMono());
```

Comment:
clean this up. Also, format this document, there shouldn't be two spaces after `instanceof TransactionalState`